File: project_euler_py/problem_048.py
"""The series, 1**1 + 2**2 + 3**3 + ... + 10**10 = 10405071317.

Find the last ten digits of the series, 1**1 + 2**2 + 3**3 + ... + 1000**1000.
"""

num = 0
for n in range(1, 1001):
    num += (n**n)

answer = str(num)[-10:]

print(answer)

# Answer: 9110843700
# Time: 0m0.035s

File: project_euler_rb/problem_048.rb
# The series, 1^1 + 2^2 + 3^3 + ... + 10^10 = 10405071317.

# Find the last ten digits of the series, 1^1 + 2^2 + 3^3 + ... + 1000^1000.

a = (1..1000).map do |x|
  x = x**x
end

puts a.sum.to_s.split('').last(10).join

# Complete!
# Answer = 9110846700
